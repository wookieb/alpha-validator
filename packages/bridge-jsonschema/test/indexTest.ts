import {byJsonSchema} from "@src/index";

describe('index', () => {
	const validation = byJsonSchema({
		type: 'object',
		properties: {
			name: {
				type: 'object',
				required: ['firstName', 'lastName'],
				additionalProperties: false,
				properties: {
					firstName: {
						type: 'string'
					},
					lastName: {
						type: 'string'
					},
				}
			},
			age: {
				type: 'number'
			}
		}
	});

	const VALID_INPUT = {
		name: {firstName: 'Tom', lastName: 'Foo'},
		age: 24
	};

	it('returns input object on success', () => {
		const result = validation(VALID_INPUT);

		expect(result.value)
			.toEqual(VALID_INPUT);
	});


	it('returns fail when invalid', () => {
		const result = validation({
			name: {lastName: 'Foo'},
			age: 24
		});

		expect(result.value)
			.toMatchSnapshot();
	});
});
