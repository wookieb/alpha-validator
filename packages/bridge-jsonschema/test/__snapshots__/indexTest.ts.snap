// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index returns fail when invalid 1`] = `
ViolationsList {
  "violations": Array [
    Object {
      "message": "must have required property 'firstName'",
      "path": Array [
        "name",
      ],
    },
  ],
}
`;
